lib/cog/repository/bundles.ex

```diff
@@ -631,6 +631,40 @@ defmodule Cog.Repository.Bundles do
     |> preload
   end
 
+  defp update_version!(version, params) do
+    params = Map.merge(params, Map.take(params["config_file"], ["author", "homepage"]))
+
+    version
+    |> BundleVersion.changeset(params)
+    |> Repo.update!
+    |> preload
+  end
+
+  defp new_or_updated_version!(bundle, params) do
+    # Check for an old version
+    version = params["config_file"]["version"]
+    result =
+      Repo.one(from bv in BundleVersion,
+               where: bv.bundle_id == ^bundle.id,
+               where: bv.version == ^version)
+
+    case result do
+      nil ->
+        # Create a new version if there isn't one
+        new_version!(bundle, params)
+      old_version ->
+        # If there is a version update it and then delete any commands and
+        # templates associated with it.
+        new_version = update_version!(old_version, params)
+        |> Repo.preload(:templates)
+
+        Enum.each(new_version.commands, &Repo.delete!/1)
+        Enum.each(new_version.templates, &Repo.delete!/1)
```

Comment:
That's not clear from this... you're calling `delete` on the commands and templates from `new_version`.